src/tsickle.ts

```diff
@@ -1542,11 +1542,20 @@ class Annotator extends ClosureRewriter {
       return;
     }
 
-    // Emit the reverse mapping of foo[foo.BAR] = 'BAR'; lines for number enums.
-    if (enumType === 'number') {
-      for (const member of node.members) {
-        const memberName = member.name.getText();
+    // Emit the reverse mapping of foo[foo.BAR] = 'BAR'; lines for number enum members
+    for (const member of node.members) {
+      const memberName = member.name.getText();
+      const memberValue = this.typeChecker.getConstantValue(member);
+      // Initializer expressions are not allowed for string enum members so this routine
+      // assumes that any enum members with an initializer have type number.
+      if (typeof memberValue === 'string') {
```

Comment:
should we invert this and rather be conservative, i.e. check for `typeof memberValue !== 'number'`?

Reply:
👍